src/compiler/scala/tools/nsc/transform/LambdaLift.scala

```diff
@@ -503,6 +503,11 @@ abstract class LambdaLift extends InfoTransform {
             }
 
             treeCopy.ValDef(tree, mods, name, tpt1, factoryCall)
+          } else if(sym.enclClass.isSubClass(DelayedInitClass) && !mods.hasFlag(CAPTURED)) {
```

Comment:
`mods.hasFlag(CAPTURED)` has no effect, as that flag is only set on symbols, never on modifiers.

Reply:
`CAPTURED` might be the wrong name. Some flag with the same value is set for syms where `isCapturedVariable` reports `true` but they are actually constructor params (if I remember correctly), not local vals.